### PR TITLE
[agent] Add documentation for Agent encryption at rest

### PR DESCRIPTION
**Pull request by andrewvc**
Based on https://github.com/elastic/elastic-agent/pull/398 and a discussion with @cmacknz .

It's a somewhat common ask from synthetics users.

### CODE REVIEW

File: docs/en/ingest-management/elastic-agent/elastic-agent-encryption.asciidoc [hunk 1]
```diff
@@ -0,0 +1,29 @@
+[elastic-agent-encryption]
+= Elastic Agent Configuration Encryption
+
+It is important for users to understand the security model of Elastic Agent and how it handles sensitive values in integration configurations.
+At a high level, Elastic Agent receives configuration data from Fleet Server over an encrypted connection, persists those secrets in an encrypted form on disk.
+This persistence is necessary to allow agents to continue to operate, even in the event of a loss of connectivity to the Fleet server.
+
+The entirety of the Fleet Agent Policy is encrypted at rest, but is recoverable if the user has access to both the encrypted configuration data and the associated key.
```
Comment (andrewvc) [inline docs/en/ingest-management/elastic-agent/elastic-agent-encryption.asciidoc:8]: Can we double-confirm this @cmacknz ?

Comment (cmacknz) [reply docs/en/ingest-management/elastic-agent/elastic-agent-encryption.asciidoc:8]: @aleksmaus would be good to have you review this as well.

File: docs/en/ingest-management/elastic-agent/elastic-agent-encryption.asciidoc [hunk 2]
```diff
@@ -0,0 +1,29 @@
+[elastic-agent-encryption]
+= Elastic Agent Configuration Encryption
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/elastic-agent-encryption.asciidoc:2]:
```suggestion
= {agent} configuration encryption
```

File: docs/en/ingest-management/elastic-agent/elastic-agent-encryption.asciidoc [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+It is important for users to understand the security model of Elastic Agent and how it handles sensitive values in integration configurations.
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/elastic-agent-encryption.asciidoc:4]:
```suggestion
It is important for you to understand the {agent} security model and how it handles sensitive values in integration configurations.
```

File: docs/en/ingest-management/elastic-agent/elastic-agent-encryption.asciidoc [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+At a high level, Elastic Agent receives configuration data from Fleet Server over an encrypted connection, persists those secrets in an encrypted form on disk.
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/elastic-agent-encryption.asciidoc:5]:
```suggestion
At a high level, {agent} receives configuration data from {fleet-server} over an encrypted connection, and persists the configuration in an encrypted form on disk.
```
Do you mean persists the configuration? "persists those secrets" makes it sound like only sensitive info is encrypted.

File: docs/en/ingest-management/elastic-agent/elastic-agent-encryption.asciidoc [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+This persistence is necessary to allow agents to continue to operate, even in the event of a loss of connectivity to the Fleet server.
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/elastic-agent-encryption.asciidoc:6]:
```suggestion
This persistence allows agents to continue to operate even if they are unable to connect to the {fleet-server}.
```

File: docs/en/ingest-management/elastic-agent/elastic-agent-encryption.asciidoc (hunk 1 above)
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/elastic-agent-encryption.asciidoc:8]:
```suggestion
The entire Fleet Agent Policy is encrypted at rest, but is recoverable if you have access to both the encrypted configuration data and the associated key.
```

File: docs/en/ingest-management/elastic-agent/elastic-agent-encryption.asciidoc [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+The entirety of the Fleet Agent Policy is encrypted at rest, but is recoverable if the user has access to both the encrypted configuration data and the associated key.
+The key material in stored in an OS-dependent manner as described below.
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/elastic-agent-encryption.asciidoc:9]:
```suggestion
The key material is stored in an OS-dependent manner as described in the following sections.
```

File: docs/en/ingest-management/elastic-agent/elastic-agent-encryption.asciidoc [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+[discreet]
+== Darwin (Mac OS)
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/elastic-agent-encryption.asciidoc:12]:
```suggestion
== Darwin (macOS)
```

File: docs/en/ingest-management/elastic-agent/elastic-agent-encryption.asciidoc [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+Key material is stored in the system keychain. The value is stored as is without any additional transformations.
+
+[discreet]
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/elastic-agent-encryption.asciidoc:16]:
```suggestion
[discrete]
```

File: docs/en/ingest-management/elastic-agent/elastic-agent-encryption.asciidoc [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+== Windows
+
+Configuration data is encrypted with https://learn.microsoft.com/en-us/dotnet/standard/security/how-to-use-data-protection[DPAPI] `CryptProtectData` with `CRYPTPROTECT_LOCAL_MACHINE``. 
+Additional entropy is derived from crypto/rand bytes stored in the `.seed` file.
+Configuration data is stored as separate files where the name of the file is SHA256 hash of the key and the content of the file is encrypted with DPAPI data.
+The security of key data relies on file system permissions. Only the Administrator should be able to access the file.
+
+[discreet]
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/elastic-agent-encryption.asciidoc:24]:
```suggestion
[discrete]
```

File: docs/en/ingest-management/elastic-agent/elastic-agent-encryption.asciidoc [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+== Darwin (Mac OS)
+
+Key material is stored in the system keychain. The value is stored as is without any additional transformations.
+
+[discreet]
+== Windows
+
+Configuration data is encrypted with https://learn.microsoft.com/en-us/dotnet/standard/security/how-to-use-data-protection[DPAPI] `CryptProtectData` with `CRYPTPROTECT_LOCAL_MACHINE``. 
+Additional entropy is derived from crypto/rand bytes stored in the `.seed` file.
+Configuration data is stored as separate files where the name of the file is SHA256 hash of the key and the content of the file is encrypted with DPAPI data.
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/elastic-agent-encryption.asciidoc:21]:
```suggestion
Configuration data is stored as separate files, where the name of the file is a SHA256 hash of the key, and the content of the file is encrypted with DPAPI data.
```

File: docs/en/ingest-management/elastic-agent/elastic-agent-encryption.asciidoc [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+The security of key data relies on file system permissions. Only the Administrator should be able to access the file.
+
+[discreet]
+== Linux
+
+The encryption key is derived from crypto/rand bytes that is written to the `.seed` file after PBKDF2 transformation.
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/elastic-agent-encryption.asciidoc:27]:
```suggestion
The encryption key is derived from crypto/rand bytes stored in the `.seed` file after PBKDF2 transformation.
```
Edited to be consistent with previous sentence about the seed file.

File: docs/en/ingest-management/elastic-agent/elastic-agent-encryption.asciidoc [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+Configuration data is stored as separate files where the name of the file is SHA256 hash of the key and the content of the file is AES256-GSM encrypted.
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/elastic-agent-encryption.asciidoc:28]:
```suggestion
Configuration data is stored as separate files, where the name of the file is a SHA256 hash of the key, and the content of the file is AES256-GSM encrypted.
```

**Review by dedemorton — COMMENTED** [review]
Just some minor copy edits. Thank you for adding this!

File: docs/en/ingest-management/elastic-agent/elastic-agent-encryption.asciidoc (hunk 8 above)
Comment (dedemorton) [reply docs/en/ingest-management/elastic-agent/elastic-agent-encryption.asciidoc:16]: IMO asciidoc's naming here is unfortunate. As it is, you end up with:

image

**Review by dedemorton — APPROVED** [review]
Doc changes LGTM.

**Comment by andrewvc** [comment]
Great, just need to get final signoff from @aleksmaus

**Review by aleksmaus — APPROVED** [review]
LGTM

**Comment by andrewvc** [comment]
@Mergifyio backport 8.6

**Pull request merged**